pkgs/build-support/trivial-builders.nix

```
@@ -76,9 +76,17 @@ rec {
     # name of the resulting derivation
     # TODO(@Artturin): enable strictDeps always
     }: buildCommand:
+    let
+      buildCommand' = ''
+        runHook preBuild
+        ${buildCommand}
+        runHook postBuild
+      '';
+    in
     stdenv.mkDerivation ({
       enableParallelBuilding = true;
-      inherit buildCommand name;
+      inherit name;
+      buildCommand = buildCommand';
```

```suggestion
      buildCommand = ''
        runHook preBuild

        ${buildCommand}

        runHook postBuild
      '';
```
This is not working?

---

Can you be more clear? I haven't been able to figure out how to test it without rebuilding the world.

---

I applied your suggestions and was able to test it in the repl:

```
nix-repl> t = outputs.legacyPackages.x86_64-linux.runCommand "test" { postBuild = "echo 'test1' >> $out"; } "touch $out; echo 'test' >$out"
nix-repl> :b t
❯ cat /nix/store/ygakm6wkgr1758fqv2qw44grgqnqs06c-test
test
test1
```